.travis.yml

```diff
@@ -21,6 +21,8 @@ jdk:
 
 before_install:
   # Remove outdated settings.xml from Travis builds. Workaround for https://github.com/travis-ci/travis-ci/issues/4629
+  - sed -i 's/^orcid\.clientid.*/orcid.clientid='$ORCID_CLIENTID'/g' dspace-api/src/test/data/dspaceFolder/config/local.cfg
+  - sed -i 's/^orcid\.clientsecret.*/orcid.clientsecret='$ORCID_CLIENTSECRET'/g' dspace-api/src/test/data/dspaceFolder/config/local.cfg
```

Comment:
You shouldn't need to do any of this at all.  The DSpace configuration system will also read values set as environment variables automatically.  See https://github.com/DSpace/DSpace/blob/master/dspace/config/config-definition.xml#L29-L37

So, instead of all this logic, you can simply set environment variables named `orcid.clientid` and `orcid.clientsecret`, and those should override any defaults in `local.cfg`.